do not remove all networks before adding a network | Add Disconnect f…

psppalash opened the pull request:
…unction to remove all networks

costanic reviewed (changes requested):
* Missing reason for change in commit message.
* Can this be split into 2 commits, one that removes the RemoveAll() call, and one that adds the Disconnect() function.
* I don't see where the Disconnect function is being called.
* Can this change be upstreamed?

psppalash committed:
add Disconnect function

Disconnect function removes all configured networks from the interface

psppalash committed:
do not remove all networks before adding a network in Connect function

This had to be done because of an issue encountered while testing the wifi Connect function on real hardware. The Connect function was removing all networks and then trying to add a network. Wpa_supplicant service on receiving the command for removing all networks was also bringing the interface down for some reason (not sure what the reason is) and then the command for adding a network was hence failing since the interface was not up

costanic commented:
> Yes, I will split it into 2 commits and add the reason to the commit message. I don't think we need to upstream this change because we changed the code for our use-case.

I still recommend pushing a PR upstream so that the maintainer can decide, or at least be aware.  Best case, the maintainer accepts the change and we no longer have to maintain a fork, which is well worth the effort IMO.

psppalash commented:
> > Yes, I will split it into 2 commits and add the reason to the commit message. I don't think we need to upstream this change because we changed the code for our use-case.
> 
> I still recommend pushing a PR upstream so that the maintainer can decide, or at least be aware. Best case, the maintainer accepts the change and we no longer have to maintain a fork, which is well worth the effort IMO.
> 
> The more I think about this, the more it seems like a generic problem, not just our use case. If the Connect() function always brings down all networks, and this has a side-effect of bringing down the interface, then wouldn't any user of this library encounter the same issue?

I had accidentally opened an upstream PR (https://github.com/mark2b/wpa-connect/pull/1) before creating this PR. The original author left a comment. I closed that PR thinking that the changes are specific to our use-case. But I can reopen it if you want.
Could it be that the problem is hardware specific?

psppalash committed:
add contribution header

psppalash commented:
@costanic Can I merge this PR? This is blocking https://github.com/armPelionEdge/maestro/pull/117

costanic commented:
Yes.  Please continue working to get this upstreamed.
